app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  # The function creates a comment for a post and updates the comment counter for the post.
+  def create
+    @post = Post.find(params[:id])
+    @comment = current_user.comments.build(comment_params.merge(post: @post))
+    if @comment.save
+      @comment.update_comments_count
+      redirect_to user_post_path(@post), notice: 'Comment added successfully.'
+    else
+      render :show
+    end
+  end
```

Comment:
- [ ] Kindly make sure to solve the error that is not allowing the user to create a new comment. Kindly fix that issue so the user can create comments successfully

  ### ERROR DISPLAYED
  ![image](https://github.com/Benawi/Blog-App/assets/87197412/f43b0e55-6759-43e6-9f17-2f822694ac64)